### ISSUE TITLE
Raise guarded union if-pattern declarations

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1991]
## Summary

Union declaration patterns in simple `if` statements are raised, and expression-bodied `pet is Cat cat && ...` cases are covered. A guarded property/declaration form in an `if` statement still leaks the lowered `Value as Cat` shape.

## Shape

```csharp
if (pet is Cat { Age: > 3 } cat)
    return cat.Name;
if (pet is Dog dog)
    return dog.Name;
return "other";
```

Current output keeps the first arm lowered:

```csharp
Cat cat = pet.Value as Cat;
if (cat is not null && cat.Age > 3)
{
    return cat.Name;
}
if (pet is Dog dog)
{
    return dog.Name;
}
return "other";
```

## Acceptance criteria

- Add a preview-SDK fixture for guarded union `if` declaration/property-pattern statements.
- Raise the first arm to `if (pet is Cat cat && cat.Age > 3)` or equivalent property-pattern spelling.
- Preserve existing simple union declaration pattern and expression-pattern tests.
- Preserve non-union `Value` lookalikes.
